src/msi/msi_token.go

```diff
@@ -139,6 +150,9 @@ func InvokeTokenRequest(resource string, msiIdentity string) (int, []byte, error
 	}
 	msi_parameters := url.Values{}
 	msi_parameters.Add("resource", resource)
+	if bypassCache {
+		msi_parameters.Add("bypass_cache", "true")
```

Comment:
`msi_parameters.Set` should be preferred, since Add would append to the existing Header, but Set would overwrite.
I am not sure how would Azure behave for some of these parameters if someof these parameters are twice.
Same for `req.Header.Add`.

Reply:
Its a URL Parameter not a header.